src/sentry/integrations/vsts/integration.py

```diff
@@ -288,8 +288,13 @@ def build_integration(self, state):
             },
         }
 
-        if not IntegrationModel.objects.filter(
-                provider='vsts', external_id=account['AccountId']).exists():
+        try:
+            subscription = IntegrationModel.objects.get(
+                provider='vsts',
+                external_id=account['AccountId']
+            ).metadata['subscription']
+            assert 'id' in subscription and 'secret' in subscription
+        except (IntegrationModel.DoesNotExist, KeyError, AssertionError):
```

Comment:
Do we want to recreate the subscription when one of `id` or `secret` are missing? (if that's a possibility)

Reply:
Oh. @mnoble I have no idea. I guess so. This really starts to feel like error territory, but I think that's an ok fix. Good catch